recipes/CapabilityPatternNode/src/Main.purs

```diff
@@ -0,0 +1,43 @@
+module CapabilityPatternNode.Main where
+
+import Prelude
+
+import App.Production.Sync (runApp, Environment) as Sync
+import App.Production.Async (runApp, Environment) as Async
+import App.Test (runApp, Environment) as Test
+import App.Application (program)
+import Effect (Effect)
+import Effect.Aff (launchAff_)
+import Effect.Class (liftEffect)
+import Effect.Class.Console (log)
+import Test.Assert (assert)
+
+
+-- | Layer 0 - Running the `program` in three different contexts
+main :: Effect Unit
+main = launchAff_ do
+  -- we can do aff-ish things here with Async/ProductionA version
+  result <- Async.runApp program { asyncEnv: "recipes/CapabilityPatternNode/async.txt" }
+  -- ...also able to do synchronous things (within Aff) using liftEffect
+  liftEffect $ mainSync { productionEnv: "recipes/CapabilityPatternNode/sync.txt" }
+  liftEffect $ mainTest { testEnv: "Test" }
+  pure unit
+
+
+
+-- Three different "main" functions for three different scenarios
+mainSync :: Sync.Environment -> Effect Unit
+mainSync env = do
+  result <- Sync.runApp program env
+  pure unit
+
+mainTest :: Test.Environment -> Effect Unit
+mainTest env = do
+  assert $ (Test.runApp program env) == "succeeds"
+  log "first test succeeded, now a failing test which will crash"
+  assert $ (Test.runApp program env) == "failing test"
```

Comment:
It looks like the failing case is really just testing the `assert` function and not `runApp`.
Is the plan to use different input for the second ("failing") test? In that case, we could probably still use an affirmative assert that expects the returned error value. Then we'd avoid crashing in CI.
We could alternatively just stick with the single passing test case.

Reply:
We could also rename `nodeSupported.md` to `nodeSupportedSkipCI.md` and that will no longer run this program during CI. I think there is value showing both successful and failing tests.

Reply:
the assert could be clearer - you run the same program but expect a different result and you get a failure

i think a recipe of this size and complexity is right at the edge of what can reasonably be in a cookbook but at the same time, AFAICT, something like this really would be the minimal case for a real world application structure so i think it does belong (unless and until there's like Cookbook 2: Advanced PureScript Recipes)

for that reason it does seem as if the failing test belongs

Reply:
> I think there is value showing both successful and failing tests.

Yes, but the current form doesn't add any value. It's pretty much the same as:
```purs
assert $ "success" == "failing test"
```

We should instead do something like:
```purs
 assert $ Test.runApp bad_program bad_env == "~expected output for bad input"
```
Where the assert is still happy and does not crash.

Reply:
so `bad_program bad_env` just runs and produces a string  "~expected output for bad input"?
or is the tilde significant in some way?  it may be that i'm ignorant of how test is supposed to work somehow but i'm not seeing how that shows us running the same program in all situations (production, production aff, test).

Reply:
Going for ["approximate"](https://en.wikipedia.org/wiki/Tilde#Common_use) with that tilde. The actual error should be more precise, so could be an `Either`, or another ADT.

I don't know what should to do for testing this outside of Test, or if that's even what should be done. I mostly just wanted to point out that tests should be designed to avoid `assert` crashes (even when checking for expected failures).

Reply:
Just noticed this message:
> you run the same program but expect a different result and you get a failure

That sounds like a test for purity. I don't know if that's exactly what we need to be testing for, but if we do wish to test for that, it would be more correct to just re-run the initial test with the same expected output. That has two benefits:
- Assert won't crash. CI still works. We can continue to test for more things.
- It's a stronger check. For example, if `runApp` was not pure, it would be indistinguishable from the pure version with that test in almost all (infinite - 1) possible outcomes. The only distinguishable case is when `runApp` happens to return "failing test", but you'd still get the identical behavior (as far as `assert` is concerned) to the pure version by impurely returning anything else, e.g. "whatever25235".

Reply:
How about we just comment out the last two lines? Someone who clones the repo and tries it out themselves can verify that it fails.

Reply:
That sounds the easiest path - obviously we can revisit it and make it better in the future, too.